### PR TITLE
Cls 482 2 generate reminders for task assigned to me from others

### PR DESCRIPTION
**Pull request by PaulWheatcroft**
## Description of change

Send email notifications when someone assigned. Task to someone else.

### Testing

> # You have been assigned to task: ((task_title)) by ((modified_by))
> 
> Company: ((company_name))
> Date due: ((task_due_date))
> 
> [View task](((task_url)))

### Checklist

* [x] Has this branch been rebased on top of the current `main` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [ ] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/main/docs/CONTRIBUTING.md) for more guidelines.

### CODE REVIEW

File: datahub/task/tasks.py [hunk 1]
```diff
@@ -170,21 +175,58 @@ def update_task_reminder_email_status(email_notification_id, reminder_ids):
...
+        adviser
+    ).email_reminders_enabled
+
+    if send_email and is_user_feature_flag_active(
+        ADVISER_TASKS_USER_FEATURE_FLAG_NAME,
+        adviser,
+    ):
+        notify_adviser_by_rq_email(
+            adviser=adviser,
+            template_identifier=settings.TASK_REMINDER_STATUS_TEMPLATE_ID,
```
Comment (chopkinsmade) [inline datahub/task/tasks.py:222]: Need to add the new setting identifier to the vault for all environments

File: datahub/task/tasks.py [hunk 2]
```diff
@@ -22,25 +24,27 @@
...
     )
 
 
-def create_task_reminder_subscription_task(adviser):
+def create_task_reminder_subscription_task(adviser_id):
     """
     Creates a task reminder subscription for an adviser if the adviser doesn't have
     a subscription already.
     """
+    adviser = Advisor.objects.get(id=str(adviser_id))
```
Comment (chopkinsmade) [inline datahub/task/tasks.py:43]: Adviser can be None here, need to add a conditional check

File: datahub/task/signals.py [hunk 3]
```diff
@@ -22,14 +22,17 @@ def delete_investment_project_task_delete(sender, instance, **kwargs):
...
     passed to the task for processing to add task reminder subscriptions
     """
-    advisers = instance.advisers.all()
-    for adviser in advisers:
-        schedule_create_task_reminder_subscription_task(adviser)
-        schedule_create_task_assigned_to_me_from_others_subscription_task(adviser)
+    task = kwargs.pop('instance', None)
+    pk_set = kwargs.pop('pk_set', None)
+    action = kwargs.pop('action', None)
+    if action == 'post_add':
```
Comment (chopkinsmade) [inline datahub/task/signals.py:35]: I think this if statement should be expanded to include a check for task not being None, and the same for the pk_set

**Review by chopkinsmade — APPROVED** [review]

File: datahub/task/signals.py (hunk 3 above)
Comment (PaulWheatcroft) [reply datahub/task/signals.py]: Applied if action == 'post_add' and pk_set != None and task != None:

Comment (PaulWheatcroft) [reply datahub/task/signals.py]: Applied if action == 'post_add' and pk_set != None and task != None:

**Review by backslashbaker — APPROVED** [review]

**Pull request merged**